fastapi_async_langchain/responses/retrieval_qa.py

```diff
@@ -20,37 +18,10 @@ def _create_chain_executor(
         chain: BaseRetrievalQA, inputs: Union[Dict[str, Any], Any]
     ) -> Callable[[Send], Awaitable[Any]]:
         async def wrapper(send: Send):
-            if not isinstance(
-                chain.combine_documents_chain.llm_chain.llm.callback_manager,
-                AsyncCallbackManager,
-            ):
-                raise TypeError(
-                    "llm.callback_manager must be an instance of AsyncCallbackManager"
-                )
-            for (
-                handler
-            ) in chain.combine_documents_chain.llm_chain.llm.callback_manager.handlers:
-                if isinstance(handler, AsyncStreamingResponseCallback):
-                    chain.combine_documents_chain.llm_chain.llm.callback_manager.remove_handler(
-                        handler
-                    )
-                    break
-            chain.combine_documents_chain.llm_chain.llm.callback_manager.add_handler(
-                AsyncLLMChainStreamingCallback(send=send)
+            llm_callback = AsyncLLMChainStreamingCallback(send=send)
+            retrieval_callback = AsyncRetrievalQAStreamingCallback(send=send)
+            return await chain.acall(
+                inputs=inputs, callbacks=[llm_callback, retrieval_callback]
```

Comment:
need to test and see how this works. if all good, we can change how `AsyncRetrievalQAStreamingCallback` works. it can inherit from `AsyncLLMChainStreamingCallback` so we don't add multiple callbacks if not needed.